### PR TITLE
Encodes the uploaded file's headers

### LINKED ISSUE #38080: Encoding of headers for ActionDispatch::Http::UploadedFile
When I upload a file with a `file_field` where the filename has UTF-8 characters (i.e. `Município_SP.shp`), an `ActionDispatch::Http::UploadedFile` object is created. The object looks like this:
```#<ActionDispatch::Http::UploadedFile:0x00007fe0f0598c00
     @content_type="application/octet-stream",
     @headers="Content-Disposition: form-data; name=\"aoi[vector_file]\"; filename=\"Munici\xCC\x81pio_SP.shp\"\r\n" + "Content-Type: application/octet-stream\r\n",
     @original_filename="Município_SP.shp",
     @tempfile=#<File:/tmp/RackMultipart20191223-20-jmab4k.shp>>
```

`original_filename` is encoded `UTF-8` (see #869). However, the `headers` are encoded in `ASCII-8BIT`.  I think `headers` should be given the same encoding as `original_filename`, which could come from the form parameter `"utf8"=>"✓"`.

Why does this matter? I instrument my code with `ActiveSupport::Notifications.subscribe` subscribed to `process_action.action_controller` and I log all my requests and parameters as JSON blobs.  Calling `payload.to_json`  raises the exception:

    Encoding::UndefinedConversionError: "\xCC" from ASCII-8BIT to UTF-8

This seems to be coming from `UploadedFile#headers`.

```
$ rails --version 
Rails 5.2.4
$ ruby --version
ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-linux]
```

### PR DESCRIPTION
**Pull request by brunoarueira**
### Summary

This pull request address the issue #38080 

Fixes #38080

### CODE REVIEW

File: actionpack/lib/action_dispatch/http/upload.rb [hunk 1]
```diff
@@ -28,6 +28,8 @@ def initialize(hash) # :nodoc:
         @tempfile = hash[:tempfile]
         raise(ArgumentError, ":tempfile is required") unless @tempfile
 
+        @content_type      = hash[:type]
```
Comment (marivaldo) [inline actionpack/lib/action_dispatch/http/upload.rb:31]:
```suggestion
        @content_type = hash[:type]
```

File: actionpack/lib/action_dispatch/http/upload.rb [hunk 2]
```diff
@@ -40,8 +42,17 @@ def initialize(hash) # :nodoc:
           @original_filename = nil
         end
 
-        @content_type      = hash[:type]
-        @headers           = hash[:head]
+        if hash[:head]
+          @headers           = hash[:head].dup
```
Comment (marivaldo) [inline actionpack/lib/action_dispatch/http/upload.rb:46]:
```suggestion
          @headers = hash[:head].dup
```

**Review by marivaldo — CHANGES_REQUESTED** [review]

**Comment by brunoarueira** [comment]
@marivaldo I've applied the suggested changes!

**Comment by brunoarueira** [comment]
@zzak @rafaelfranca Someone could review this PR? 😄

**Review by zzak — COMMENTED** [review]
🤔 This approach seems reasonable to me, but I'm not sure I have enough context to give it a 👍

**Comment by jessetuglu** [comment]
Hey any status on this PR?

**Comment by brunoarueira** [comment]
It's depending on the rails core team to review yet

**Comment by marcelolx** [comment]
+1

**Comment by brunoarueira** [comment]
waiting for the rails team to review 😃 

ping @rafaelfranca

**Comment by brunoarueira** [comment]
still waiting for the rails team 😃

**Pull request merged**